tutorials/haskell/snap-web-app/tutorial.md

```diff
@@ -0,0 +1,490 @@
+---
+title: How to create a web application in Haskell with Snap
+published: 2017-01-30
+ghc: 8.0.1
+lts: 7.18
+libraries: snap-core-1.0.1.0 snap-server-1.0.1.1 heist-1.0.1.0 HDBC-sqlite3-2.3.3.1    
+language: haskell
+description: In this tutorial we show you how to build a simple TODO web app using Snap framework
+author-name: Omar García
+---
+
+## Introduction
+
+[Snap](http://snapframework.com/) is a simple and [fast](http://snapframework.com/blog/2010/11/17/snap-0.3-benchmarks)
+web development framework for the Haskell programming language with a clean
+[monads](https://wiki.haskell.org/Monad) oriented design. Snap apps are often
+composed by [snaplets](http://snapframework.com/docs/tutorials/snaplets-tutorial),
+auto-contained pieces of reusable functionalities that can be glued together to build a website.
+
+In this tutorial we'll show you how to build a simple TODO web app using Snap,
+[SQLite](https://www.sqlite.org/), and [Stack](http://docs.haskellstack.org/en/stable/README/)
+as our Haskell build tool. The target app should deliver the following features:
+
```

Comment:
...
Display the tasks on the TODO list
Create a new task and add it to the TODO list
Mark a task as completed
Delete a task from the TODO list
...